Regression in 1.7: Dev mode auto-compile fails in MapStruct with "Attempt to recreate a file for type ..."

**Describe the bug**
I am getting this strange error during auto-compile while trying to start dev mode in one of our modules:
```
[INFO] --- quarkus-maven-plugin:1.7.0.CR2:dev (default-cli) @ middleware-patient-api-pro-fe ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 54 source files to C:\Develop\someproject\middleware\_develop-2\someproject-middleware\patient\api\pro-fe\target\classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /C:/Develop/someproject/middleware/_develop-2/someproject-middleware/patient/api/pro-fe/src/main/java/com/somecompany/someproject/middleware/patient/api/pro_fe/data/mapper/CaseFileApiMapper.java:[26,8] Internal error in the mapping processor: java.lang.Rat org.codehaus.plexus.classworlds.launcat org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)  Caused by: javax.annotation.processing.FilerException: Attempt to recreate a file for type com.somecompany.someproject.middleware.patient.api.pro_fe... 49 more  Casat jdk.compiler/com.sun.tools.javac.processing.JavacFiler.createSourceFile(JavacFiler.java:426)  stence(at org.mapstruct.ap.internal.processor.MapperRenderingProcessor.createSourceFile(Mappe[INFO] 1 error
```
I have no clue why this is so horribly garbled. `mvn -e ...` doesn't help much but at least reveals which type was attempted to recreate.
```
Attempt to recreate a file for type com.somecompany.someproject.middleware.patient.api.pro_fe.data.mapper.CaseFileApiMapperImpl
    at org.mapstruct.ap.internal.processor.MapperRen(... garbled, I suppose MapperRenderingProcessor)
```

This is _not_ flaky, btw. Only the affected "MapperImpl" can fluctuate (we have multiple mappers).

Dev mode in another module does work, but there we do not use MapStruct.

**The very same code works with 1.6.1.Final!**

**Expected behavior**
Should just compile (if needed) without errors.

**Actual behavior**
Build is aborted.


**To Reproduce**
Steps to reproduce the behavior:
1. git clone clone https://github.com/famod/quarkus-quickstarts
2. git switch issue-11253-mapstruct-vs-prepare 
4. cd getting-started
5. should pass: mvn clean verify (or install)
6. should fail: mvn quarkus:dev

Note: This reproducer uses MapStruct 1.4.0.Beta3. Downgrading to 1.3.1.Final does not change anything.

**Configuration**
n/a

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `MINGW64_NT-10.0-18363 XXX 3.0.7-338.x86_64 2019-11-21 23:07 UTC x86_64 Msys`
 - Output of `java -version`: `OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)`
 - GraalVM version (if different from Java): n/a
 - Quarkus version or git rev: 1.7.0.CR1 or 1.7.0.CR2
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

**Additional context**
`mvn clean compile quarkus:dev` works, which is no wonder since compile is skipped in this case.

I found two commits that may have something to do with it (the first one being the "hottest" candidate):
- https://github.com/quarkusio/quarkus/commit/93e1ba6f97fa2714fe67c42d8d197f0210437100#diff-c007939524ea2a3177991ef913abf5df
- https://github.com/quarkusio/quarkus/commit/3d7e64230f97a1772fb6bb58b929a2471e5a5eb2#diff-c007939524ea2a3177991ef913abf5df

I did not yet have the time to debug this.

--- a/devtools/maven/src/main/java/io/quarkus/maven/CodeGenMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CodeGenMojo.java
@@ -106,8 +106,6 @@ public class CodeGenMojo extends AbstractMojo {
                     ? buildDir.toPath().resolve("generated-test-sources")
                     : buildDir.toPath().resolve("generated-sources");
 
-            sourceRegistrar.accept(generatedSourcesDir);
-
             QuarkusClassLoader deploymentClassLoader = curatedApplication.createDeploymentClassLoader();
             List<CodeGenData> codeGens = CodeGenerator.init(deploymentClassLoader,
                     Collections.singleton(sourcesDir),

